### ISSUE TITLE
Examples don't work when installation instructions are followed (.local)

### ISSUE THREAD

**Issue by MLopez-Ibanez** [esa/pagmo2#312]
The documentation suggests:

```
cmake -DCMAKE_INSTALL_PREFIX=~/.local .
```

However, none of the examples will work this way because GCC does not search in ~/.local by default. Every example must be amended to use something like:

```bash
g++ -c -O2 -DNDEBUG -std=c++11 nsga2.cpp -I $HOME/.local/include -o nsga2.o
g++ -O2 -DNDEBUG -std=c++11 nsga2.o  -L $HOME/.local/lib  -lpagmo -Wl,-R$HOME/.local/lib  -o nsga2
```

**Comment by bluescarni** [comment]
CC #285.

**Comment by bluescarni** [comment]
Being addressed in #340.

**Comment by bluescarni** [comment]
Fixed.
